[BUG] Unable to build project due to invalid Hadoop dependency paths and DNS resolution errors

### What is the bug?
After cloning the repository, the build fails when trying to resolve Hadoop dependencies. This issue is similar to elastic/elasticsearch-hadoop#2197 but affects the OpenSearch-hadoop project.

### How can one reproduce the bug?
1. Clone the repository
2. Run `./gradlew build`

### What is the expected behavior?
The project should build successfully with all dependencies resolved.

### What is your host/environment?
- Java: 17.0.10
- Gradle: 8.12.1

```
❯ sw_vers
ProductName:		macOS
ProductVersion:		15.5
BuildVersion:		24F74
```

### Do you have any additional context?
Build fails with dependency resolution errors:
```
FAILURE: Build failed with an exception.

What went wrong:
Could not determine the dependencies of task ':qa:kerberos:hadoopFixture#datanode.extract'.


Could not resolve all files for configuration ':qa:kerberos:downloadHadoop#3.3.6'.
Could not resolve hadoop.common:hadoop-3.3.6:hadoop-3.3.6.
Required by:
project :qa:kerberos
> Could not resolve hadoop.common:hadoop-3.3.6:hadoop-3.3.6.
> Could not get resource 'https://repo.spring.io/plugins-release-local/hadoop/common/hadoop-3.3.6/hadoop-3.3.6/hadoop-3.3.6-hadoop-3.3.6.pom'.
> Could not GET 'https://repo.spring.io/plugins-release-local/hadoop/common/hadoop-3.3.6/hadoop-3.3.6/hadoop-3.3.6-hadoop-3.3.6.pom'. Received status code 401 from server:
> Could not resolve hadoop.common:hadoop-3.3.6:hadoop-3.3.6.
> Could not get resource 'https://apache.osuosl.org/hadoop/common/hadoop-3.3.6/hadoop-3.3.6.tar.gz'.
> Could not HEAD 'https://apache.osuosl.org/hadoop/common/hadoop-3.3.6/hadoop-3.3.6.tar.gz'.
> apache.osuosl.org: nodename nor servname provided, or not known
```

### Root Causes

1. **Invalid dependency coordinates**: The generated coordinates `hadoop.common:hadoop-3.3.6:hadoop-3.3.6` are malformed
2. **Inaccessible repositories**: 
   - Spring repository (`repo.spring.io/plugins-release-local`) returns 401
   - OSUOSL mirror (`apache.osuosl.org`) has DNS resolution issues
3. **Incorrect Ivy pattern**: The artifact pattern doesn't match the actual Hadoop distribution structure

Related to

- elastic/elasticsearch-hadoop#2197

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 ### Removed
 
 ### Fixed
+- Fixed build failures when downloading Apache project dependencies (Hadoop, Hive, Spark)
 
 ### Security
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,7 +11,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 ### Removed
 
 ### Fixed
-- Fixed build failures when downloading Apache project dependencies (Hadoop, Hive, Spark)
+- Fixed build failures when downloading Apache project dependencies (Hadoop, Hive, Spark) ([#595](https://github.com/opensearch-project/opensearch-hadoop/pull/595))
 
 ### Security
 

--- a/buildSrc/src/main/groovy/org/opensearch/hadoop/gradle/fixture/hadoop/HadoopFixturePlugin.groovy
+++ b/buildSrc/src/main/groovy/org/opensearch/hadoop/gradle/fixture/hadoop/HadoopFixturePlugin.groovy
@@ -40,7 +40,7 @@ import org.gradle.api.artifacts.repositories.IvyPatternRepositoryLayout
 
 class HadoopFixturePlugin implements Plugin<Project> {
 
-    private static final String APACHE_MIRROR = "https://apache.osuosl.org/"
+    private static final String APACHE_MIRROR = "https://archive.apache.org/dist/"
 
     static class HadoopFixturePluginExtension {
         private NamedDomainObjectContainer<HadoopClusterConfiguration> clusters
@@ -81,10 +81,9 @@ class HadoopFixturePlugin implements Plugin<Project> {
         repositoryHandler.add(repositoryHandler.ivy({IvyArtifactRepository ivyArtifactRepository ->
             ivyArtifactRepository.setUrl(APACHE_MIRROR)
             ivyArtifactRepository.patternLayout({IvyPatternRepositoryLayout ivyPatternRepositoryLayout ->
-                // We use this pattern normally and break the regular tradition of a strictly numerical version
-                // because Hive does not provide a reasonable artifact name that makes a more robust pattern
-                // reasonable (it has a very unorthodox layout)
-                ivyPatternRepositoryLayout.artifact("[organization]/[module]/[revision].[ext]")
+                ivyPatternRepositoryLayout.artifact("[module]/common/[module]-[revision]/[module]-[revision].[ext]")
+                ivyPatternRepositoryLayout.artifact("[module]/[module]-[revision]/apache-[module]-[revision]-bin.[ext]")
+                ivyPatternRepositoryLayout.artifact("spark/spark-[revision]/spark-[revision]-bin-hadoop3.[ext]")
                 ivyPatternRepositoryLayout.setM2compatible(true)
             })
             ivyArtifactRepository.metadataSources({IvyArtifactRepository.MetadataSources metadataSources ->

--- a/buildSrc/src/main/groovy/org/opensearch/hadoop/gradle/fixture/hadoop/services/HadoopServiceDescriptor.groovy
+++ b/buildSrc/src/main/groovy/org/opensearch/hadoop/gradle/fixture/hadoop/services/HadoopServiceDescriptor.groovy
@@ -77,7 +77,8 @@ class HadoopServiceDescriptor implements ServiceDescriptor {
 
     @Override
     String getDependencyCoordinates(ServiceConfiguration configuration) {
-        return "hadoop.common:hadoop-${configuration.getVersion()}:${artifactName(configuration)}@tar.gz"
+        Version version = configuration.getVersion()
+        return "apache:hadoop:${configuration.getVersion()}@tar.gz"
     }
 
     @Override

--- a/buildSrc/src/main/groovy/org/opensearch/hadoop/gradle/fixture/hadoop/services/HiveServiceDescriptor.groovy
+++ b/buildSrc/src/main/groovy/org/opensearch/hadoop/gradle/fixture/hadoop/services/HiveServiceDescriptor.groovy
@@ -74,7 +74,7 @@ class HiveServiceDescriptor implements ServiceDescriptor {
 
     @Override
     String getDependencyCoordinates(ServiceConfiguration configuration) {
-        return "hive:hive-${configuration.getVersion()}:${artifactName(configuration)}@tar.gz"
+        return "apache:hive:${configuration.getVersion()}@tar.gz"
     }
 
     @Override

--- a/buildSrc/src/main/groovy/org/opensearch/hadoop/gradle/fixture/hadoop/services/SparkYarnServiceDescriptor.groovy
+++ b/buildSrc/src/main/groovy/org/opensearch/hadoop/gradle/fixture/hadoop/services/SparkYarnServiceDescriptor.groovy
@@ -77,7 +77,7 @@ class SparkYarnServiceDescriptor implements ServiceDescriptor {
 
     @Override
     String getDependencyCoordinates(ServiceConfiguration configuration) {
-        return "spark:spark-${configuration.getVersion()}:${artifactName(configuration)}@tgz"
+        return "apache:spark:${configuration.getVersion()}@tgz"
     }
 
     @Override

